Tensile/KernelWriter.py

```diff
@@ -472,41 +503,88 @@ def makeSubIterSchedule(self, kernel, localReadCode, iteration, pointerLWCode, p
       iterCode.addCode(waitCode)
 
       # interleave pack code
-      packItem = packCode.flatitems()
-      packCount = 0
-      packInstPerIter = (1 + kernel["MIWaveTile"][0] * kernel["MIWaveTile"][1]) * (kernel["MatrixInstK"] // 2)
-      while packItem:
-        if packCount < packInstPerIter:
-          item = packItem.pop(0)
-          iterCode.addCode(item)
-          packCount = packCount + 1
-        else:
-          break
-
-      if packCount == 0:
-        tmpVgpr = self.vgprPool.checkOut(1)
-        iterCode.addInst("v_mov_b32 ","v%u"%(tmpVgpr),"0x0","valu operation to have different priority")
-        self.vgprPool.checkIn(tmpVgpr)
+      instPerPack = 1 if kernel["ProblemType"]["DataType"].isBFloat16() else 2
+      packItems = []
+      for iui in range(kernel["InnerUnroll"]):
+        packINtems = [ [] for j in range(max(self.numReadsIterCoalescedA,self.numReadsIterCoalescedB)) ]
+        packA = packCode.findNamedItem("packA_I%s"%(iui))
+        packB = packCode.findNamedItem("packB_I%s"%(iui))
+        # In case localReadDo not generate pack Module
+        # and findNamedItem will return None type
+        # TODO: let all type have pack Module
+        if not packA:
+          packA = Code.Module()
+        packAItems = packA.flatitems()
+        if not packB:
+          packB = Code.Module()
+        packBItems = packB.flatitems()
+        if packAItems:
+          for j in range(self.numReadsIterCoalescedA):
+            for n in range(instPerPack):
+              packINtems[j].append(packAItems.pop(0))
+        if packBItems:
+          for j in range(self.numReadsIterCoalescedB):
+            for n in range(instPerPack):
+              packINtems[j].append(packBItems.pop(0))
+        while packAItems:
+          for j in range(self.numReadsIterCoalescedA):
+            for n in range(instPerPack):
+              packINtems[j].append(packAItems.pop(0))
+        while packBItems:
+          for j in range(self.numReadsIterCoalescedB):
+            for n in range(instPerPack):
+              packINtems[j].append(packBItems.pop(0))
+        for j in range(max(self.numReadsIterCoalescedA,self.numReadsIterCoalescedB)):
+          packItems += packINtems.pop(0)
 
-      iterCode.addInst("s_setprio ","3","Raise priority while processing macs")
       macIterItem = macIterCode.flatitems()
       # pop the first code which is s_nop 1 for packing
       item = macIterItem.pop(0)
-      iterCode.addCode(item)
 
-      while macIterItem:
+      numMfmaPerIter = kernel["MIWaveTile"][0] * kernel["MIWaveTile"][1] * kernel["InnerUnroll"]
+      curPackIdx = 0
+      packAIdx = 0
+      packBIdx = 0
+
+      for i in range(numMfmaPerIter):
```

Comment:
how about pack A and B interleave so that can be A[0:1], B[0:1] before first mfma, and A[2:3] befor second mfma, rather than A[0:1], A[2:3], B[0:1] before first mfma.  Can consider in next optimization stage.